pkg/api/dashboard_public.go

```diff
@@ -84,16 +84,28 @@ func (hs *HTTPServer) QueryPublicDashboard(c *models.ReqContext) response.Respon
 		return response.Error(http.StatusBadRequest, "invalid panel ID", err)
 	}
 
+	dashboard, err := hs.dashboardService.GetPublicDashboard(c.Req.Context(), web.Params(c.Req)[":uid"])
+	if err != nil {
+		return response.Error(http.StatusInternalServerError, "could not fetch dashboard", err)
+	}
+
+	publicDashboard, err := hs.dashboardService.GetPublicDashboardConfig(c.Req.Context(), dashboard.OrgId, dashboard.Uid)
+	if err != nil {
+		return response.Error(http.StatusInternalServerError, "could not fetch public dashboard", err)
+	}
+
 	reqDTO, err := hs.dashboardService.BuildPublicDashboardMetricRequest(
```

Comment:
I think we have some duplication of effort in BuildPublicDashboardMetricRequest. We're doing parameter substitution for time here and in GetPublicDashboard and also doing publicDashboard.isEnabled check inside BuildPublicDashboardMetricRequest. I think we can probably flatten these methods.